File: src/main/java/com/dh/ondot/schedule/domain/PlaceHistory.java
package com.dh.ondot.schedule.domain;

import java.time.Instant;

public record PlaceHistory(
        Long memberId,
        String title,
        String roadAddress,
        Double longitude,
        Double latitude,
        Instant searchedAt
) {
    public static PlaceHistory of(
            Long memberId, String title, String roadAddress,
            Double longitude, Double latitude
    ) {
        return new PlaceHistory(
                memberId,
                title,
                roadAddress,
                longitude,
                latitude,
                Instant.now()
        );
    }
}

File: src/main/kotlin/com/dh/ondot/schedule/domain/PlaceHistory.kt
package com.dh.ondot.schedule.domain

import java.time.Instant

data class PlaceHistory(
    val memberId: Long,
    val title: String,
    val roadAddress: String,
    val longitude: Double,
    val latitude: Double,
    val searchedAt: Instant,
) {
    companion object {
        fun of(
            memberId: Long, title: String, roadAddress: String,
            longitude: Double, latitude: Double,
        ): PlaceHistory = PlaceHistory(
            memberId = memberId,
            title = title,
            roadAddress = roadAddress,
            longitude = longitude,
            latitude = latitude,
            searchedAt = Instant.now(),
        )
    }
}
